### PR TITLE
Add azure logo to readme

### PR DESCRIPTION
**Pull request by TanZng**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR adds the Azure logo to the README.md 🥳 

**Special notes for your reviewer**:
Using that logo is ok?

I took the image from https://github.com/Azure profile picture

On https://capz.sigs.k8s.io/

| Before | After ✨  |
|---|---|
|image |image|

On GitHub README :octocat: 

| Before | After ✨  |
|---|---|
|image|image|

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests

**Release note**:
```
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [![Go Report Card](https://goreportcard.com/badge/kubernetes-sigs/cluster-api-provider-azure)](https://goreportcard.com/report/kubernetes-sigs/cluster-api-provider-azure)
 
-<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png"  width="100">
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png"  width="100"><a href="https://azure.microsoft.com/en-us/overview/open-source/"><img src="https://azure.microsoft.com/svghandler/information-protection/?width=200&height=110" alt="Powered by Azure"></a>
```
Comment (CecileRobertMichon) [inline README.md:5]: there is actually a new Azure logo since May 7th https://azure.microsoft.com/en-us/blog/a-fluent-new-look-for-the-azure-icon/

Comment (TanZng) [reply README.md:5]: Oh okay! Do u know where I can found a png or svg of the new logo :D?

Comment (TanZng) [reply README.md]: Updated now with new azure logo 😄  thanks!

Comment (CecileRobertMichon) [reply README.md]: Awesome. What do you think about adding a bit of space between the two logos so it's a little more spread out?

Comment (CecileRobertMichon) [reply README.md]: also, where did you end up getting the new logo from? Poked around a bit and couldn't find the svg file anywhere

Comment (TanZng) [reply README.md]: I also couldn't find the svg :c so I took the image from the Github Azure profile picture https://github.com/Azure

Comment (TanZng) [reply README.md]: Space between added 👍🏻 !

**Comment by CecileRobertMichon** [comment]
Thanks for the PR @TanZng! I need to check with the Microsoft legal team to make sure this is an appropriate use of the logo. 

/hold

**Comment by CecileRobertMichon** [comment]
Unfortunately, the legal team recommended not to include any trademarks or logos in an open source project so I have to close this. 

/close

**Pull request closed without merging**